core/src/main/java/bio/terra/pearl/core/service/datarepo/DataRepoExportService.java

```diff
@@ -110,24 +114,26 @@ public void createDataset(StudyEnvironment studyEnv, String datasetName, String
                 .studyEnvironmentId(studyEnv.getId())
                 .build();
 
+        Dataset createdDataset = datasetService.create(dataset);
+
         DataRepoJob job = DataRepoJob.builder()
                 .studyEnvironmentId(studyEnv.getId())
                 .status(response.getJobStatus().getValue())
                 .datasetName(datasetName)
+                .datasetId(createdDataset.getId())
                 .tdrJobId(response.getId())
                 .jobType(JobType.CREATE_DATASET)
                 .build();
 
         dataRepoJobService.create(job);
-        datasetService.create(dataset);
     }
 
-    public String uploadCsvToAzureStorage(UUID studyEnvironmentId, String datasetName) {
+    public String uploadCsvToAzureStorage(UUID studyEnvironmentId, UUID datasetId) {
         ExportOptions exportOptions = new ExportOptions(false, false, false, ExportFileFormat.TSV, null);
 
         //Even though this is actually formatted as a TSV, TDR only accepts files ending in .csv or .json.
         //In the DataRepoClient call, we specify that the CSV delimiter is "\t", which will make it all work fine.
-        String blobName = datasetName + "_" + studyEnvironmentId + "_" + Instant.now() + ".csv";
+        String blobName = datasetId.toString() + "_" + studyEnvironmentId + "_" + Instant.now() + ".csv";
```

Comment:
Preparing for the future `deleteDataset` feature- a datasetName could theoretically be re-used so I don't want to cross wires with blobNames